rpc_jobs/bump_snapshots.yml

```diff
@@ -0,0 +1,134 @@
+        }
+        // NOTE(mattt): We re-clone rpc-gating here because the existing clone
+        // has cruft in it which we don't want to have committed in our
+        // subsequent PR
+        dir("rpc-gating-master") {
+          git branch: 'master', url: 'https://github.com/rcbops/rpc-gating'
+
+          def componentData = readFile "${env.WORKSPACE}/releases/components/rpc-openstack.yml"
+          def rpcoReleases = readYaml text: componentData
+
+          List releases = []
+
+          for ( r in rpcoReleases['releases'] ) {
+            if (r['series'] == 'newton') {
+              for ( s in ['trusty', 'xenial'] ) {
+                def image = "${s}_loose_artifacts-swift"
+                releases << [series: r['series'],
+                             latest: r['versions'][0]['version'],
+                             previous: r['versions'][1]['version'],
+                             image: image]
+              } // for
+            } else {
+              releases << [series: r['series'],
+                           latest: r['versions'][0]['version'],
+                           previous: r['versions'][1]['version'],
+                           image: 'xenial_no_artifacts-swift' ]
+            } // if
+          } // for
+
+          def parallelBuilds = [:]
+
+          for ( r in releases ) {
+            def release = r
+            println release
+
+            parallelBuilds["${release['series']}-${release['image']}"] = {
+              def buildResult = build(
+                job: 'PM_rpc-gating-master-snapshot-snapshot-test',
+                wait: true,
+                propagate: false,
+                parameters: [
+                  [
+                    $class: "StringParameterValue",
+                    name: "RPC_GATING_BRANCH",
+                    value: env.RPC_GATING_BRANCH,
+                  ],
+                  [
+                    $class: "StringParameterValue",
+                    name: "BRANCH",
+                    value: env.RPC_GATING_BRANCH,
+                  ],
+                  [
+                    $class: "StringParameterValue",
+                    name: "IMAGE",
+                    value: "rpc-${release['latest']}-${release['image']}",
+                  ]
+                ]
+              )
+              if (buildResult.result != "SUCCESS") {
+                println "Build failed, no updates will take place for image rpc-${release['latest']}-${release['image']}"
+              }
+              release['result'] = buildResult.result
+            } // parallelBuilds
+          } // for
+
+          parallel parallelBuilds
+
+          for ( release in releases ) {
+            if (release['result'] == "SUCCESS") {
+              sh """#!/bin/bash -xe
+                sed -i "s/rpc-${release['previous']}-${release['image']}/rpc-${release['latest']}-${release['image']}/g" rpc_jobs/*.yml
+              """
+            }
```

Comment:
The failures get ignored, is something else responsible for those? The commit message suggests bit rot testing but it doesn't look like any action gets taken when there is an error.

Reply:
They now get ignored because we released a new version with a missing snapshot image which resulted in Bump-Snapshot-Images failing every time it ran. I don't want to error any time a build fails -- as that will result in a bunch of noise when stuff doesn't boot up in time, etc. Any suggestions on what I should do here?

Reply:
Can't the missing snapshot be created after the release? That strikes me as the kind of failure we would want to know about.

With this set up there is nothing to prompt anyone if there are problems with the snapshots, so projects could stop getting updated but we'd never know unless someone manually review things.

Maybe you could have it log an issue if it has failed a certain number of times in a row or if the image, or a newer one, still isn't being used after a specified number of days from its release.

Reply:
@git-harry got a change I'm working on that involves checking for consecutive failures and alerting if there are 7 or more consecutive failures. I'll need to do a bit more testing on this tomorrow but hopefully this satisfies your concern.